codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/AsyncClientClass.java

```diff
@@ -503,18 +491,14 @@ private boolean hasStreamingV4AuthOperations() {
                 .anyMatch(this::shouldUseAsyncWithBodySigner);
     }
 
-    private MethodSpec batchMangerMethod() {
-        String scheduledExecutor = "executorService";
-
-        BatchManagerMethod config = model.getCustomizationConfig().getBatchManagerMethod();
-        ClassName returnType = PoetUtils.classNameFromFqcn(config.getAsyncReturnType());
-
-        return MethodSpec.methodBuilder(BatchManagerMethod.METHOD_NAME)
-                         .returns(returnType)
-                         .addModifiers(Modifier.PUBLIC)
-                         .addAnnotation(Override.class)
-                         .addStatement("return $T.builder().client(this).scheduledExecutor($N).build()",
-                                       returnType, scheduledExecutor)
-                         .build();
+    private void addBatchManagerMethodAndScheduledExecutorIfNeeded(Builder classBuilder, boolean hasScheduledExecutor) {
```

Comment:
Would it make sense to separate them so that waiter can reuse the `addScheduledExecutorIfNeeded` and `hasScheduledExecutor` will be managed by `addScheduledExecutorIfNeeded` only

Reply:
Separated method into `addScheduledExecutorIfNeeded` and `addBatchManagerMethod`